lib/APMSpan.js

```diff
@@ -21,31 +21,41 @@ import { APMSpanContext } from './APMSpanContext';
 class APMSpan {
     constructor(tracer, fields) {
         this._tracer = tracer;
-        this._spanContext = APMSpan._createSpanContext(fields.childOf);
+        if (fields.childOf && fields.childOf instanceof APMSpan) {
+            fields.childOf = fields.childOf.context();
+        }
+
+        this._initSpanContext(fields.childOf);
 
         this._operationName = fields.operationName;
         this._tags = fields.tags || {};
         this._startMillis = fields.startTime || nowMillis();
         this._endMillis = null;
         this._logs = [];
-        this._sampled = tracer.getSampler().isSampled(this);
 
         this._finished = false;
     }
 
-    static _createSpanContext(parentSpanOrContext) {
+    _initSpanContext(parentSpanContext) {
         const id = generateSpanId();
```

Comment:
No, meant interaction id - however it is being used for correlation purposes within the model. But on the wire it is an interaction id - but the key is shortened to 'id'.